### ISSUE TITLE
Javascript: errors prevent full-page loader from being removed

### ISSUE THREAD

**Issue by toconnell** [toconnell/kdm-manager#265]
From email:

> [597a93828740d90ad4a50bac] has submitted an error report!
> 
> The report goes as follows:
> I'm getting a permanent loading pinwheel on every page, even though I can see that everything loaded properly behind the pinwheel? Can't click on anything

And:

> Hey Tim,
> 
> I get the pinwheel spinning forever on my mobile device. Standard iphone, I think I get the error on the Settlement Sheet, Survivor Sheet, and the Campaign page. I can see the information loaded behind the pinwheel, but it never seems to go away and it whites out the backgrounds. Checked the issue on my PC and it seems like there are no problems there. Thanks

**Comment by toconnell** [comment]
I think this is also to blame on the now-deprecated _apiService_ factory. 

Once I refactor the Survivor Sheet to work like the Settlement Sheet and one-shot its init method, I'll check this out on Windows with a few browsers and see if I can reproduce.

tldr; I think multiple API hits during _initialize()_ and _reinitialize()_ are to blame for this.

**Comment by toconnell** [comment]
Yeah, this seems to go away with the new routine for one-shot initialization. I'm leaving it open until final QA on this release.

**Comment by toconnell** [comment]
I saw something during the last deploy that I think is related to this:

```
[2017-09-09 13:34:44] INFO:     [toconnell@tyrannybelle.com (565f3d67421aa95c4af1e230)] authorization check failed: Signature has expired!
[2017-09-09 13:34:50] INFO:     Converted principles from names (legacy) to handles for settlements object 'Gravestone' [58d05a518740d90e66f7252d]
[2017-09-09 13:34:50] INFO:     settlements object 'Gravestone' [58d05a518740d90e66f7252d] settlement modified during normalization! Saving changes...
[2017-09-09 13:34:50] INFO:     Saved settlements object 'Gravestone' [58d05a518740d90e66f7252d] to mdb.settlements successfully!
[2017-09-09 13:34:55] ERROR:    Incorrectly formatted token!
[2017-09-09 13:34:55] ERROR:    Incorrectly formatted token!
[2017-09-09 13:34:55] ERROR:    Incorrectly formatted token!
```

The actual code is in the API's User class methods:

```
153 def token_to_object(request, strict=True):
154     """ Processes the "Authorization" param in the header and returns an http
155     response OR a user object. Requires the application's initialized JWT to
156     work. """
157
158     # first, get the token or bail
159
160     auth_token = request.headers.get("Authorization", None)
161     if auth_token is None:
162         logger.error("Authorization header missing!")
163         return utils.http_401
164
165     # now, try to decode the token and get a dict
166
167     try:
168         decoded = jwt.decode(auth_token, secret_key, verify=strict)
169         user_dict = dict(json.loads(decoded["identity"]))
170         return User(_id=user_dict["_id"]["$oid"])
171     except jwt.DecodeError:
172         logger.error("Incorrectly formatted token!")
173     except Exception as e:
174         logger.exception(e)
175
176     return utils.http_401
```

tl;dr I think there's something going on in the Legacy Webapp where it sends a badly/incorrectly formatted Authorization header (e.g. with an old JWT or something).

**Comment by toconnell** [comment]
These are popping up periodically as well:

```
[2017-09-09 15:15:04] ERROR:    Error attempting to process cookie!
[2017-09-09 15:15:04] ERROR:
[2017-09-09 15:15:06] ERROR:    Error attempting to process cookie!
[2017-09-09 15:15:06] ERROR:
```

**Comment by toconnell** [comment]
This may also be related to https://github.com/toconnell/kdm-manager/issues/283 (#283)

**Comment by toconnell** [comment]
Closing this. Haven't heard about it in donkeys' years: assuming it was fixed back when we started doing single init calls from the angular app.
